pymc3/sampling.py

```diff
@@ -1919,17 +1932,15 @@ def sample_prior_predictive(
         prior_vars = (
             get_default_varnames(model.unobserved_RVs, include_transformed=True) + model.potentials
         )
-        vars_ = [var.name for var in prior_vars + prior_pred_vars]
-        vars = set(vars_)
+        vars_: Iterable[str] = [var.name for var in prior_vars + prior_pred_vars]
     elif vars is None:
-        vars = var_names
-        vars_ = vars
-    elif vars is not None:
+        assert var_names is not None  # help mypy
+        vars_ = var_names
+    elif var_names is None:
         warnings.warn("vars argument is deprecated in favor of var_names.", DeprecationWarning)
         vars_ = vars
     else:
         raise ValueError("Cannot supply both vars and var_names arguments.")
-    vars = cast(TIterable[str], vars)  # tell mypy that vars cannot be None here.
```

Comment:
only `vars_` could be used after this point, any reason to assign to `vars`?

Reply:
I think the reason is in the first `if` branch: `vars = set(vars_)`. Looks like the `for` in Line 1945/1956 is supposed to iterate over each value exactly once.